src/scanpy/preprocessing/_scale.py

```diff
@@ -203,25 +217,13 @@ def scale_array(
     # do the clipping
     if max_value is not None:
         logg.debug(f"... clipping at max_value {max_value}")
-        if isinstance(X, DaskArray) and issparse(X._meta):
-
-            def clip_set(x):
-                x = x.copy()
-                x[x > max_value] = max_value
-                if zero_center:
-                    x[x < -max_value] = -max_value
-                return x
-
-            X = da.map_blocks(clip_set, X)
+        if isinstance(X, DaskArray):
+            clip = clip_set if issparse(X._meta) else clip_array
+            X = X.map_blocks(clip, max_value=max_value, zero_center=zero_center)
+        elif issparse(X):
+            X.data = clip_array(X.data, max_value=max_value, zero_center=False)
         else:
-            if isinstance(X, DaskArray):
-                X = X.map_blocks(
-                    clip_array, max_value=max_value, zero_center=zero_center
-                )
-            elif issparse(X):
-                X.data = clip_array(X.data, max_value=max_value, zero_center=False)
-            else:
-                X = clip_array(X, max_value=max_value, zero_center=zero_center)
+            X = clip_array(X, max_value=max_value, zero_center=zero_center)
```

Comment:
This works with zappy? is that the fix?

Reply:
Ah sorry; context.

No, the logic is untouched, only the `maybe_dask_process_context` is new.

This PR is part of me figuring out the whole “numba abort” situation, and this logic refactor is a side effect. I should probably retitle this PR.

Reply:
"This" being this line that was the `else` with `clip_array` which appears to only be for `numpy`